rpk: add confirmation prompt to iotune to force execution

r-vasquez opened the pull request:
## Cover letter

iotune tuner checked if the io-config file was already present before executing, which means if the user wanted to rerun iotune (after a fix or a mount change), it would have to delete the file manually.                                                                
                                                                         
Now the user will have to confirm via a CLI prompt if they want to proceed with an Iotune execution and we warn them that this will delete previous iotune results.                                                
                                                                         
In the future, we should change this overwriting approach in the redpanda-iotune binary to something like appending each test result to a list. 

Part of #6149, the rest of the ticket should be handled in redpanda-iotune.

## Backport Required

<!-- Specify which branches this should be backported to, e.g.: -->
- [ ] papercut/not impactful enough to backport
- [ ] v22.2.x

## UX changes
Assuming that we already ran iotune once and still have the `/etc/redpanda/io-config.yaml`:

Before:
```
# This won't execute iotune, just skip it without telling the user what's going on
$ rpk iotune
Starting iotune...
IO configuration file stored as '/etc/redpanda/io-config.yaml'
```

Now:
```
$ rpk iotune
? IO config file detected, do you want to continue? doing this will overwrite previous results. No
IO tune canceled.

$ rpk iotune
? IO config file detected, do you want to continue? doing this will overwrite previous results. Yes
Starting iotune...
...
```

and this will force the iotune execution without prompt.
```
$ rpk iotune --no-confirm
```

## Release notes
### Improvements

* rpk now will prompt the user for confirmation when executing iotune multiple times.

twmb commented on src/go/rpk/pkg/cli/cmd/iotune/iotune.go:
```diff
@@ -39,17 +40,31 @@ func NewCommand(fs afero.Fs) *cobra.Command {
-			err = execIoTune(fs, evalDirectories, outputFile, duration, timeout)
-			out.MaybeDie(err, "error during IoTune execution: %v", err)
+			if exists, _ := afero.Exists(fs, outputFile); exists && !force {
+				out.Die("unable to run iotune: IO config file already exists in: %q; use --force if you want to overwrite its content", outputFile)
```
The main existing flag we for forced runs is `--no-confirm` in both acl deletion and partition movement canceling. What do you think about switching this to a prompt if the file exists, and then using --no-confirm here?

r-vasquez replied:
Done

r-vasquez commented:
[Force Push](https://github.com/redpanda-data/redpanda/compare/e943335f668bd092ea9d2482f4c830856eedb31c..2de401cf7fd297f2ac8500e1c8d870e821cd25da): 

- Changed the --force flag to a confirmation prompt. Updated commit message and PR description

0x5d commented on src/go/rpk/pkg/cli/cmd/iotune/iotune.go:
```diff
@@ -39,17 +40,35 @@ func NewCommand(fs afero.Fs) *cobra.Command {
+			)
+			fmt.Println("Starting iotune...")
+			result := tuner.Tune()
+			out.MaybeDie(result.Error(), "error during IoTune execution: %v", result.Error())
```
Please make the spelling of `iotune` consistent. There's `IO tune`, `iotune` and `IoTune`. Since the binary is `iotune`, I think we should stick with that.

r-vasquez commented:
[Force Push](https://github.com/redpanda-data/redpanda/compare/8f4be92997b505777fb309f6314d121d0ca2bbd1..7dc446603c19ff2b35a0994e0ca81e1533989a5b):

- spelling of `iotune` standardized.

twmb commented on src/go/rpk/pkg/cli/cmd/iotune/iotune.go:
```diff
@@ -85,27 +104,6 @@ func NewCommand(fs afero.Fs) *cobra.Command {
 			"fraction and a unit suffix, such as '300ms', '1.5s' or '2h45m'. "+
 			"Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'",
 	)
+	command.Flags().BoolVar(&noConfirm, "no-confirm", false, "Disable confirmation prompt when executing iotune multiple times")
```
`Disable confirmation prompt if the iotune file already exists`

twmb commented on src/go/rpk/pkg/cli/cmd/iotune/iotune.go:
```diff
@@ -39,17 +40,35 @@ func NewCommand(fs afero.Fs) *cobra.Command {
-			err = execIoTune(fs, evalDirectories, outputFile, duration, timeout)
-			out.MaybeDie(err, "error during IoTune execution: %v", err)
+			if exists, _ := afero.Exists(fs, outputFile); exists && !noConfirm {
+				confirmed, err := out.Confirm("IO configuration file detected, do you want to continue? doing this will overwrite previous results.")
```
`Overwrite existing configuration file at %q?`

r-vasquez commented:
[Force Push](https://github.com/redpanda-data/redpanda/compare/7dc446603c19ff2b35a0994e0ca81e1533989a5b..b1f7a25ded64a159f93382bc120b4ecb431bdff7):

Changes in help flag description and confirmation prompt.

twmb commented on src/go/rpk/pkg/cli/cmd/iotune/iotune.go:
```diff
@@ -39,17 +40,35 @@ func NewCommand(fs afero.Fs) *cobra.Command {
+			result := tuner.Tune()
+			out.MaybeDie(result.Error(), "error during iotune execution: %v", result.Error())
+
+			fmt.Printf("IO configuration file stored as %q\n", outputFile)
```
as => at
